src/Components/WebAssembly/Build/src/Tasks/BrotliCompressBlazorApplicationFiles.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Build
+{
+    public class BrotliCompressBlazorApplicationFiles : ToolTask
+    {
+        private const string DotNetHostPathEnvironmentName = "DOTNET_HOST_PATH";
+
+        [Required]
+        public ITaskItem StaticWebAsset { get; set; }
+
+        [Required]
+        public string BlazorBrotliPath { get; set; }
+
+        private string _dotnetPath;
+
+        private string DotNetPath
+        {
+            get
+            {
+                if (!string.IsNullOrEmpty(_dotnetPath))
+                {
+                    return _dotnetPath;
+                }
+
+                _dotnetPath = Environment.GetEnvironmentVariable(DotNetHostPathEnvironmentName);
+                if (string.IsNullOrEmpty(_dotnetPath))
+                {
+                    throw new InvalidOperationException($"{DotNetHostPathEnvironmentName} is not set");
+                }
+
+                return _dotnetPath;
+            }
+        }
+
+        protected override MessageImportance StandardErrorLoggingImportance => MessageImportance.High;
+
+        protected override string ToolName => Path.GetFileName(DotNetPath);
+
+        protected override string GenerateFullPathToTool() => DotNetPath;
+
+        protected override string GenerateCommandLineCommands()
+        {
+            var targetCompressionPath = StaticWebAsset.GetMetadata("TargetCompressionPath");
+            var sourcePath = StaticWebAsset.GetMetadata("FullPath");
+
+            var args = new StringBuilder();
+            args.Append(BlazorBrotliPath);
+            args.Append(' ');
+            args.Append(Quote(sourcePath));
+            args.Append(' ');
+            args.Append(Quote(targetCompressionPath));
+
+            return args.ToString();
+        }
+
+        private static string Quote(string path) => $"\"{path}\"";
```

Comment:
Nitpick: could be:

```suggestion
            return $"{BlazorBrotliPath} \"{sourcePath}\" \"{targetCompressionPath}\"";
        }
```

Perhaps more importantly, is it possible that `BlazorBrotliPath` needs to be quoted too? If it's an absolute path that we don't completely control, then it definitely would need to be.